### DISCUSSION TITLE
Thread hangs when connecting and consuming from RabbitMQ

### DISCUSSION THREAD

**Discussion by LamOne1** [pika/pika#1464] [Q&A]
I'm encountering a problem in our production environment that I can't replicate locally.

After processing over 2.5 million messages from our RabbitMQ queue over approximately 5 hours, the processing thread stops unexpectedly. The thread runs a straightforward function, but it seems to hang at the following segment of code:
```
       inConnection = pika.BlockingConnection(parameters)
        inChannel = inConnection.channel()
        for bat in range(batch_size):
            method_frame, header_frame, body = inChannel.basic_get(queueName) 
            if method_frame:
                inChannel.basic_ack(method_frame.delivery_tag)
                if body:
                    body=body.decode('utf-8')
            else:
                continue
                 
        inChannel.close()
        inConnection.close()
```

The `parameters` only specify the host, port, and credentials. 
Unfortunately, I'm uncertain which specific line is causing the issue.

Previously, we encountered a similar problem when new connections were added to our application without being properly closed, leading to resource exhaustion (what we believe), specifically socket-related resources. Closing those connections resolved the issue at that time. However, the problem has resurfaced after integrating a security agent that also utilizes socket connections. Although this agent appears to manage socket connections properly, the concurrent use of multiple connections might be contributing to the issue.

I would like to modify the setup so that it throws an exception instead of just hanging when a problem occurs. How can we address this issue?

I have two other threads that are connected to RabbitMQ and they operate without any issues, even when the thread that is causing problems experiences a hang-up.

The first one code snippet:
```
inConnection = pika.BlockingConnection(parameters)
inChannel = inConnection.channel()
            
num_elements = inChannel.queue_declare(queue=queueName, durable=True, \
                                                   exclusive=False, auto_delete=False).method.message_count
```

The second one:
```
outConnection = pika.BlockingConnection(parameters)
outChannel = outConnection.channel()
outChannel.basic_publish(exchange='', routing_key= queueName, body=item)
outChannel.close()
outConnection.close()
```

**package versions:**
RabbitMQ 3.8.11
Erlang 22.3
pika 1.2.0
python 3.6.13

**Comment by lukebakken** [comment]
Hello, thanks for using Pika and RabbitMQ.

There is very little I can suggest based on a couple code snippets. I have no idea how you are using Python threads, for instance.

If you can provide code that I can run to reproduce the issue, I could probably assist you. Otherwise, I don't have the time to investigate.

Note that you are using unsupported versions of all of your software.

**Comment by LamOne1** [comment]
I found the cause of my issue, it wasn't related to pika or RabbitMQ.
Thank you

**Reply by lukebakken** [reply]
Thanks for following up. If it's something that other people could learn from, please feel free to add more details!
